### PR TITLE
feat(automatic-release): Generation of automatic release

### PR DESCRIPTION
**Pull request by wardpeet**
- added release:create to create a release with standard-version
- added release:publish to publish a release to github and npm

### CODE REVIEW

**Comment by TheLarkInn** [comment]
Nice work so far. will this also perform automatic publish if the ci checks pass? Ideally we would want to automate  the deploy process also.

**Comment by SpaceK33z** [comment]
"added release:publish to publish a release to github" does this mean that it will add the changelog to the GitHub Releases page, or is this about pushing the tag only?

**Comment by wardpeet** [comment]
@TheLarkInn If you put the commands in travis it can do that yes.
@SpaceK33z only push tags, so it's a git tag release, not a manual created one.

**Comment by TheLarkInn** [comment]
So technically we'd want this to run after success for a full publish if I am understanding correctly. Maybe we can auto publish to a new tag type.

**Comment by wardpeet** [comment]
@TheLarkInn do you want to publish a prerelease or?

**Comment by TheLarkInn** [comment]
Yes. :) if ci passes let's publish to @next tag

**Comment by TheLarkInn** [comment]
@d3viant0ne has also proposed a `cron` job instead for nightlies similar to how the typescript team does it. I'm going to review this one more time, verify with @SpaceK33z and then we'll get this one merged. And then once these look great we'll go ahead and shift to the other repos. OO THIS IS SO EXCITING.

**Comment by d3viant0ne** [comment]
@SpaceK33z - The documentation around this toolset can be found here - https://webpack.js.org/development/release-process/

**Comment by SpaceK33z** [comment]
@d3viant0ne, thanks for those docs, I read them now. Do I understand it correctly that this tool generates a CHANGELOG.md entry based on the commits? If so, wouldn't it be possible to push this to the GitHub release page?

I'll also try to implement this in `webpack-dev-{middleware,server}` soon. If anyone wants to do a PR for that, that's fine too.

**Comment by d3viant0ne** [comment]
@SpaceK33z  - Yes, it's going to take a bit of scripting but the intent is to have the same mechanism update the release description. 

I would expect @wardpeet to be opening a PR at some point for server / middleware. If not, i'll take care of it. The same setup is being applied to everything to include webpack/webpack

File: package.json [hunk 1]
```diff
@@ -21,6 +22,8 @@
   "homepage": "https://github.com/webpack/tapable",
   "main": "lib/Tapable.js",
   "scripts": {
-    "test": "mocha --reporter spec"
+    "test": "mocha --reporter spec",
+    "release:create": "standard-version -i CHANGELOG.md -m \"Creating release %s\"",
+    "release:publish": "git push --follow-tags origin master; npm publish"
```
Comment (d3viant0ne) [inline package.json]: @wardpeet - Same comment as from enhanced resolve - Per the [published documentation](https://webpack.js.org/development/release-process/) we only need a single task `"release": "standard-version"`

From that we can chain commands on to publish different targets. You also don't need to specify the changelog.md nor is the message really necessary.

The publish shouldn't be done as an npm script, it eats npm publishing errors. We also don't always publish to `dist-tag`. 

In theory using an npm script could be viable for pushing tags to origin but in reality, I can copy and paste the printed command `git push --follow-tags origin master` the library spits out faster than I can type `npm run release:publish`

Comment (wardpeet) [inline package.json]: @d3viant0ne sure you can include me.

I'm open for run release. I'll fix my changes and update all PR :)
We can move from there

**Comment by sokra** [comment]
Not needed

**Pull request closed without merging**